app/uk/gov/hmrc/helptosave/actors/ThresholdMongoProxy.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.actors
+
+import akka.pattern.pipe
+import akka.actor.{Actor, Props}
+import cats.syntax.either._
+import uk.gov.hmrc.helptosave.actors.ThresholdMongoProxy._
+import uk.gov.hmrc.helptosave.repo.ThresholdStore
+
+class ThresholdMongoProxy(thresholdStore: ThresholdStore) extends Actor {
+  import context.dispatcher
+
+  def retrieveThreshold() = thresholdStore.getThreshold().value
```

Comment:
can we deal with the case class `UCThreshold` every where here which wraps the double - not sure if it makes things any better ?

Reply:
the case class is being used to read the JSON response from the connector and to write to mongo in the ThresholdStore class. In between though, it would be simpler to pass in doubles if this is not the case

Reply:
I think we are mostly doing that though

Reply:
then its no point to use the class `UCThreshold` here in the mongo store
```  private[repo] def doUpdate(amount: Double)(implicit ec: ExecutionContext): Future[Option[UCThreshold]] =```